src/python/platforms/android/adb.py

```diff
@@ -371,16 +372,33 @@ def recreate_gce_device():
   stop_gce_device()
 
   # Delete all existing images.
-  image_dir = cvd_dir
-  for image_file_path in glob.glob(os.path.join(image_dir, '*.img')):
-    shell.remove_file(image_file_path)
-
-  # Restore images from backup.
-  backup_image_dir = os.path.join(cvd_dir, 'backup')
-  for image_filename in os.listdir(backup_image_dir):
-    image_src = os.path.join(backup_image_dir, image_filename)
-    image_dest = os.path.join(image_dir, image_filename)
-    shell.copy_file(image_src, image_dest)
+  cvd_address = get_gce_address()
+  rm_cmd = 'rm -rf {cvd_dir}/*'.format(cvd_dir=cvd_dir)
+  if cvd_address:
```

Comment:
This makes it hard to distinguish between a configuration error vs a non-cuttlefish bot. 

We should error out instead, if we are on a cuttlefish bot and the address does not exist.

Also, let's extend execute_command and add a wrapper copy_file. e.g.

```python
def execute_command(..., on_cuttlefish_host=False):
   if on_cuttlefish_host:
      if environment.is_android_cuttlefish():
        cvd_address = get_gce_address()
        if not cvd_address:
          # raise an exception
    
        cmd = f'ssh {cvd_address} "{cmd}"'
      
   ...


def copy_file(..., on_cuttlefish_host=False):
  # similar
```

This will help consolidate/simplify a lot of the existing code paths.

Reply:
Done, not sure if I understand it correctly, copy file has special cases like _tar_  which is has different options on local and remote